artemis-server/src/main/java/org/apache/activemq/artemis/core/settings/impl/AddressSettings.java

```diff
@@ -1797,6 +1817,8 @@ public void encode(ActiveMQBuffer buffer) {
 
       BufferHelper.writeNullableLong(buffer, pageLimitMessages);
 
+      BufferHelper.writeNullableInteger(buffer, idCacheSize);
```

Comment:
The writing and reading of the nullable `Integer` is not in the same order so decoding will fail.

Reply:
I moved the read up to the 4th ordinal position from the bottom so they are both same now. Thank you for catching.

Reply:
Both the read and the write for the `idCacheSize` should be at the *end* otherwise the broker may choke on existing persisted address settings when upgrading.

Reply:
oh yeah it will definitely get mixed up with currently persisted messages - thanks. I put both the read and write at end now